File: assets_rebalance_frontend/src/boundaries/change-fin-asset-bank.input.ts
import { FinAssetBank } from "@/domain/fin-asset-bank";

export class ChangeFinAssetBankInput {
    id?: string; // Required (UUID is represented as a string in TypeScript)
    name: string; // Required
    routing: string; // Required
    ispb?: string | null; // Optional
    ispbName?: string | null; // Optional
    iconBase64?: string | null; // Optional

    constructor(
        name: string,
        routing: string,
        id?: string,
        ispb?: string | null,
        ispbName?: string | null,
        iconBase64?: string | null
    ) {
        this.id = id
        this.name = name;
        this.routing = routing;
        this.ispb = ispb ?? null;
        this.ispbName = ispbName ?? null;
        this.iconBase64 = iconBase64 ?? null;
    }

    toDomain(): FinAssetBank {
        const bank = new FinAssetBank(); // Initialize without parameters
        bank.id = this.id!;
        bank.name = this.name;
        bank.routing = this.routing;
        bank.ispb = this.ispb;
        bank.ispbName = this.ispbName;
        bank.iconBase64 = this.iconBase64;

        return bank;
    }
}

File: assets_rebalance_backend/src/Boundaries/ChangeFinAssetBankInput.cs
using System;
using assets_rebalance_backend.Domain;

namespace assets_rebalance_backend.Boundaries;

public class ChangeFinAssetBankInput
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public required string Name { get; set; }
    public required string Routing { get; set; }
    public string? Ispb { get; set; }
    public string? IspbName { get; set; }
    public string? IconBase64 { get; set; }

    public FinAssetBank Domain() => new()
    {
        Id = Id,
        Name = Name,
        Routing = Routing,
        Ispb = Ispb,
        IspbName = IspbName,
        IconBase64 = IconBase64
    };
}
